packages/maskbook/src/plugins/Trader/apis/bancor/index.ts

```diff
@@ -0,0 +1,76 @@
+import type {
+    SwapBancorRequest,
+    ExpectedTargetAmountResponse,
+    ExpectedSourceAmountResponse,
+    TradeTransactionCreationResponse,
+} from '../../types/bancor'
+import urlcat from 'urlcat'
+import { BANCOR_API_BASE_URL } from '../../constants'
+import { calculateMinimumReturn } from './calculateMinimumReturn'
+
+const getTargetAmount = async (request: SwapBancorRequest) => {
+    const baseUrl = BANCOR_API_BASE_URL[request.chainId!]
+    const url = urlcat(baseUrl, '/pricing/target-amount', {
+        source_dlt_type: 'ethereum',
+        source_dlt_id: request.fromToken?.address,
+        target_dlt_type: 'ethereum',
+        target_dlt_id: request.toToken?.address,
+        amount: Number(request.fromAmount!),
+    })
+    const response = await fetch(url)
+    if (response.ok) {
+        const data: ExpectedTargetAmountResponse = await response.json()
+        return data.amount
+    }
+    return undefined
+}
+
+const getSourceAmount = async (request: SwapBancorRequest) => {
+    const baseUrl = BANCOR_API_BASE_URL[request.chainId!]
+    const url = urlcat(baseUrl, '/pricing/source-amount', {
+        source_dlt_type: 'ethereum',
+        source_dlt_id: request.fromToken?.address,
+        target_dlt_type: 'ethereum',
+        target_dlt_id: request.toToken?.address,
+        amount: Number(request.toAmount!),
```

Comment:
```suggestion
        amount: Number.parseInt(request.toAmount!, 10),
```

Reply:
Is it a big number instead of a string?

Reply:
Hello @guanbinrui all values in SwapBancorRequest are decimals.

As Bancor API expects decimals, as per their docs - https://docs.bancor.network/rest-api/api-reference.

Reply:
> 
Fixed

Reply:
fixed